File: simple_linear_regression.R
#Simple Linear Regression

#Data preprocessing
dataset = read.csv("Salary_Data.csv")

#splitting the dataset into training set and test set
#install.packages('caTools')
library(caTools)
set.seed(123)
split = sample.split(dataset$Salary, SplitRatio = 2/3)
training_set = subset(dataset, split ==TRUE)
test_set = subset(dataset, split ==FALSE)

#feature scaling
# training_set[, 2:3] = scale(training_set[, 2:3])
# test_set[, 2:3] = scale(test_set[, 2:3])

#fitting simple linear regression to the training set
regressor = lm(formula = Salary ~ YearsExperience, data = training_set)

#predicting the test set results
y_pred = predict(regressor, newdata = test_set)

#visualising the training set results
install.packages('ggplot2')
library(ggplot2)
ggplot() +
  geom_point(aes(x = training_set$YearsExperience, 
                 y = training_set$Salary,
                 color = 'red')) +
  geom_line(aes(x = training_set$YearsExperience,
                 y = predict(regressor, newdata = training_set)),
                 color = 'blue') +
  ggtitle('Salary vs Experience(Training_set') +
  xlab('Years of Experience') +
  ylab('Salary')

#visualising the test set results
ggplot() +
  geom_point(aes(x = test_set$YearsExperience, 
                 y = test_set$Salary,
                 color = 'red')) +
  geom_line(aes(x = training_set$YearsExperience,
                 y = predict(regressor, newdata = training_set)),
             color = 'blue') +
  ggtitle('Salary vs Experience(Test_set') +
  xlab('Years of Experience') +
  ylab('Salary')

File: simple_linear_regression.py
#Simple linear regression
#Importing libraries
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd

 #importing datasets
dataset = pd.read_csv('Salary_Data.csv')
x = dataset.iloc[:, :-1].values
y = dataset.iloc[:, 1].values

#splitting the dataset into training set and test set
from sklearn.model_selection import train_test_split
x_train, x_test, y_train, y_test = train_test_split(x, y, test_size = 1/3, random_state = 0)

#feature scaling
"""from sklearn.preprocessing import StandardScaler
sc_x = StandardScaler()
x_train = sc_x.fit_transform(x_train)
x_test = sc_x.transform(x_test)"""

#fitting linear regression to the training set
from sklearn.linear_model import LinearRegression
regressor = LinearRegression()
regressor.fit(x_train, y_train)

#predicting the test set values
y_pred = regressor.predict(x_test)

#visualising the training set results
plt.scatter(x_train, y_train, color = 'red')
plt.plot(x_train, regressor.predict(x_train), color = 'blue')
plt.title('Salary vs Experience (Training Set)')
plt.xlabel('Years of Experience')
plt.ylabel('Salary')

#visualising the test set results
plt.scatter(x_test, y_test, color = 'red')
plt.plot(x_train, regressor.predict(x_train), color = 'blue')
plt.title('Salary vs Experience (Test Set)')
plt.xlabel('Years of Experience')
plt.ylabel('Salary')



















